src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -209,9 +209,8 @@ private boolean areAllChildrenIgnored() {
      * Returns a {@link Statement}: run all non-overridden {@code @BeforeClass} methods on this class
      * and superclasses before executing {@code statement}; if any throws an
      * Exception, stop execution and pass the exception on.
-     * @param testClass 
      */
-    public static Statement withBeforeClasses(Statement statement, TestClass testClass) {
+    protected Statement withBeforeClasses(Statement statement) {
```

Comment:
I like your proposal to introduce `withClassStatements()`. I also had something like this in mind before, but didn't brought it to an end. I will incorporate it now.

I'd like to propose to leave `withBeforeClasses(...)` (and the others) protected anyway. But the documentation should encourage to use `withClassStatements()`.

I can't call `ParentRunner.classBlock(...)` in my custom `Runner` because between invoking the children and applying the class statements, I'd like to apply `@TestRule`s. ParentRunner doesn't support these.

````
        Statement statement = childrenInvoker(notifier);

        List<TestRule> testRules = BlockJUnit4ClassRunner.getTestRules(testInstance, getTestClass());
        statement = BlockJUnit4ClassRunner.withTestRules(testRules, getDescription(), statement);

        statement = withClassStatements(statement);
        return statement;
````